html/php-pre-challenge3/index.php

```diff
@@ -1,6 +1,58 @@
 <?php
 $limit = $_GET['target'];
-
+$limit = mb_convert_kana($limit, 'a' , 'UTF-8');
+if(!is_numeric($limit) || $limit <= 0 || preg_match('/^([1-9]\d*|0)\.(\d+)?$/', $limit)) {
+    http_response_code(400);
+}
 $dsn = 'mysql:dbname=test;host=mysql';
 $dbuser = 'test';
 $dbpassword = 'test';
+try {
+    $db = new PDO($dsn,$dbuser,$dbpassword);
+} catch (PDOException $e) {
+    echo 'DB接続エラー' . $e->getMessage();
+}
+
+// DBの値を配列に格納
+$records = $db->query('select value from prechallenge3');
+$data = $records->fetchAll();
+foreach($data as $key=>$value) {
+    $num[] = $value[0];
```

Comment:
混乱の元なので配列変数は複数形にしましょう

```suggestion
    $nums[] = $value[0];
```